apis/elasticloadbalancing/v1alpha1/elb_types.go

```diff
@@ -0,0 +1,175 @@
+
+	// The instance being checked.
+	Target string `json:"target"`
+
+	// The amount of time, in seconds, during which no response means a failed health
+	// check.
+	Timeout int64 `json:"timeout"`
+
+	// The number of consecutive health check failures required before moving the
+	// instance to the Unhealthy state.
+	UnhealthyThreshold int64 `json:"unhealthyThreshold"`
+}
+
+// ELBParameters define the desired state of an AWS ELB.
+type ELBParameters struct {
+	// One or more Availability Zones from the same region as the load balancer.
+	AvailabilityZones []string `json:"availabilityZones"`
+
+	// Information about the health checks conducted on the load balancer.
+	HealthCheck *HealthCheck `json:"healthCheck,omitempty"`
+
+	// The listeners for this ELB.
+	Listeners []Listener `json:"listeners"`
+
+	// The type of a load balancer. Valid only for load balancers in a VPC.
+	// +optional
+	// +immutable
+	Scheme *string `json:"scheme,omitempty"`
+
+	// The IDs of the security groups to assign to the load balancer.
+	// +optional
+	SecurityGroups []string `json:"securityGroups,omitempty"`
+
+	// The IDs of the subnets in your VPC to attach to the load balancer. Specify
+	// one subnet per Availability Zone specified in AvailabilityZones.
+	// +optional
+	Subnets []string `json:"subnets,omitempty"`
+
+	// A list of tags to assign to the load balancer.
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// An ELBSpec defines the desired state of an ELB.
+type ELBSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ELBParameters `json:"forProvider"`
+}
+
+// ELBObservation keeps the state for the external resource
+type ELBObservation struct {
+	// Information about the EC2 instances for this ELB.
+	BackendServerDescriptions []BackendServerDescription `json:"backendServerDescriptions,omitempty"`
+
+	// The DNS name of the load balancer.
+	CanonicalHostedZoneName string `json:"canonicalHostedZoneName,omitempty"`
+
+	// The ID of the Amazon Route 53 hosted zone for the load balancer.
+	CanonicalHostedZoneNameID string `json:"canonicalHostedZoneId,omitempty"`
+
+	// The DNS name of the load balancer.
+	DNSName string `json:"dnsName,omitempty"`
+
+	// The ID of the VPC for the load balancer.
+	VPCID string `json:"vpcId,omitempty"`
+}
+
+// An ELBStatus represents the observed state of an ELB.
+type ELBStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ELBObservation `json:"atProvider"`
+}
+
+// +kubebuilder:object:root=true
+
+// An ELB is a managed resource that represents an AWS Classic Load Balancer.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
```

Comment:
Suggestion: Should we add more fields such as VPCID, or DNSName, directly in printcolumn?